File: src/creational/abstract-factory/python/classes/interfaces/user_vehicle_factory_protocol.py
from abc import ABCMeta, abstractclassmethod

from classes.interfaces.user_protocol import UserProtocol
from classes.interfaces.vehicle_protocol import VehicleProtocol


class UserVehicleFactoryProtocol(metaclass=ABCMeta):
    @staticmethod
    @abstractclassmethod
    def create_user(name: str, location: str) -> UserProtocol:
        pass

    @staticmethod
    @abstractclassmethod
    def create_vehicle(name: str) -> VehicleProtocol:
        pass

File: src/creational/abstract-factory/typescript/classes/interfaces/UserVehicleFactoryProtocol.ts
import { UserProtocol } from './UserProtocol';
import { VehicleProtocol } from './VehicleProtocol';

export interface UserVehicleFactoryProtocol {
  createUser(name: string, location: string): UserProtocol;
  createVehicle(name: string): VehicleProtocol;
}
